yt_dlp/extractor/generic.py

```diff
@@ -2203,24 +2203,25 @@ def _extract_rss(self, url, video_id, doc):
                 continue
 
             guid = try_call(lambda: it.find('guid').text)
-            if guid:
-                next_url = smuggle_url(next_url, {'force_videoid': guid})
+            next_url_new = smuggle_url(next_url, {'force_videoid': guid}) if guid else next_url
 
             def itunes(key):
                 return xpath_text(it, xpath_with_ns(f'./itunes:{key}', NS_MAP), default=None)
 
             entries.append({
                 '_type': 'url_transparent',
-                'url': next_url,
+                'url': next_url_new,
                 'title': try_call(lambda: it.find('title').text),
                 'description': xpath_text(it, 'description', default=None),
+                'uploader': xpath_text(it, 'author', default=None),
                 'timestamp': unified_timestamp(xpath_text(it, 'pubDate', default=None)),
                 'duration': parse_duration(itunes('duration')),
                 'thumbnail': url_or_none(xpath_attr(it, xpath_with_ns('./itunes:image', NS_MAP), 'href')),
                 'episode': itunes('title'),
                 'episode_number': int_or_none(itunes('episode')),
                 'season_number': int_or_none(itunes('season')),
                 'age_limit': {'true': 18, 'yes': 18, 'false': 0, 'no': 0}.get((itunes('explicit') or '').lower()),
+                'webpage_url': next_url,
```

Comment:
This is wrong. Passing `webpage_url` back to yt-dlp should return the same video id, which this change breaks

Reply:
The problem was that in RSS the final webpage URL among the metadata of the video was smuggled. This change allows the next processing steps to still make use of the information in the smuggled URL (field `url`) while keeping only the real unsmuggled webpage URL (field `webpage_url`) in the final metadata.

Reply:
> The problem was that in RSS the final webpage URL among the metadata of the video was smuggled

That is not a problem, as is intentional as mentioned in my above comment

Reply:
> Passing webpage_url back to yt-dlp should return the same video id, which this change breaks

With this change, I tested passing private unsmuggled URLs as input to yt-dlp CLI and got back the same video IDs. To help me understand what you mean, could you please give an example that breaks this?

Anyway, I think it would not be ideal to show to the users the URL of the video (possibly embedded with `--add-metadata`) smuggled, as smuggled URLs are for internal yt-dlp use.

https://github.com/yt-dlp/yt-dlp/blob/29ca408219947914b5ce1d2fa1c268a4397719f8/yt_dlp/utils.py#L2256